src/lib/storage/table.cpp

```diff
@@ -17,57 +17,112 @@
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+Column::Column(std::string name, std::string type) : name(std::move(name)), type(std::move(type)) {}
+
+Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size{target_chunk_size} {
+  // create initial chunk
+  _append_new_chunk();
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  _columns.emplace_back(name, type);
+  _append_column_to_chunks(type);
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  if (_chunks.back()->size() == _target_chunk_size) {
+    _append_new_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
-ColumnCount Table::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
+// TODO(max): write test
+void Table::emplace_chunk(std::unique_ptr<Chunk> chunk) {
+  if (row_count() == 0) {
+    // only existing chunk is empty -> replace chunk
+    _chunks[0] = std::move(chunk);
+  } else {
+    DebugAssert(_chunks.back()->size() != _target_chunk_size,
+                "Cannot emplace chunk because current last chunk is not full.");
+    _chunks.emplace_back(std::move(chunk));
+  }
```

Comment:
```suggestion
  auto last_chunk = _chunks.back();
  if(last_chunk->size() == 0) {
    last_chunk = std::move(chunk);
  } else {
    Assert(last_chunk->size() != _target_chunk_size,
                "Cannot emplace chunk because current last chunk is not full.");
    _chunks.emplace_back(std::move(chunk));
  }
```
You might have missed the case, when a chunk, which is not the first one, is still empty. This could lead to empty chunks in the middle of the table.